src/model_diagnostics/_utils/tests/test_array.py

```diff
@@ -350,3 +353,44 @@ def test_safe_index_rows(a):
         assert_array_equal(a_sub, np.array([[-99, 0], [22, 2]]))
     else:
         assert_array_equal(a_sub, np.array([-99, 22]))
+
+
+@pytest.mark.parametrize(
+    ("a", "result"),
+    [
+        (np.array([[1, 2], [3, 4]]), [0, 1]),
+        (pa_table({"b": [0, 1], "a": ["A", "B"]}), ["b", "a"]),
+        (pd_DataFrame({"b": [0, 1], "a": 0.5}), ["b", "a"]),
+        (pl.DataFrame({"b": [0, 1], "a": 0.5}), ["b", "a"]),
+    ],
+)
+def test_get_column_names(a, result):
+    """Test that get_column_names() works correctly."""
+    if isinstance(a, SkipContainer):
+        pytest.skip("Module for data container not imported.")
+    assert get_column_names(a) == result
+
+
+@pytest.mark.parametrize(
+    ("a", "result"),
+    [
+        (np.array([[1, 2], [3, 4]]), np.array([[1, 2], [3, 4]])),
+        (
+            pa_table({"a": [0, 1], "b": ["A", "B"]}),
+            pa_table({"a": [0, 1], "b": ["A", "B"]}),
+        ),
+        (pd_DataFrame({"a": [0, 1], "b": 0.5}), pd_DataFrame({"a": [0, 1], "b": 0.5})),
+        (pl.DataFrame({"a": [0, 1], "b": 0.5}), pl.DataFrame({"a": [0, 1], "b": 0.5})),
+    ],
+)
+def test_safe_copy(a, result):
+    """Test that get_second_dimension works correctly.
+
+    For simplicity, we only consider numerical values.
+    """
+    if isinstance(a, SkipContainer):
+        pytest.skip("Module for data container not imported.")
+
+    b = safe_copy(a)
+    safe_assign_column(b, values=[0, 0], column_index=1)  # has side-effects
+    assert_array_equal(a, result)
```

Comment:
Please also check that the original data has not been modified as implied by "safe".

Reply:
Solved it via:

```py
b = safe_assign_column(b, values=[0, 0], column_index=1)  # has side-effects
assert not np.array_equal(b, a)
```